gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -0,0 +1,117 @@
+use std::{fmt::Debug, sync::Arc};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fedimint_api::dyn_newtype_define;
+use tonic::{
+    transport::{Channel, Endpoint},
+    Request, Streaming,
+};
+use tracing::error;
+use url::Url;
+
+use crate::{
+    gatewaylnrpc::{
+        gateway_lightning_client::GatewayLightningClient, CompleteHtlcsRequest,
+        CompleteHtlcsResponse, GetPubKeyRequest, GetPubKeyResponse, PayInvoiceRequest,
+        PayInvoiceResponse, SubscribeInterceptHtlcsRequest, SubscribeInterceptHtlcsResponse,
+    },
+    LnGatewayError, Result,
+};
+
+#[async_trait]
+pub trait ILnRpcClient: Debug + Send + Sync {
+    /// Get the public key of the lightning node
+    async fn get_pubkey(&self) -> Result<GetPubKeyResponse>;
+
+    /// Attempt to pay an invoice using the lightning node
+    async fn pay_invoice(&self, invoice: PayInvoiceRequest) -> Result<PayInvoiceResponse>;
```

Comment:
perhaps rename this to `pay(invoice)` as in `get(amount)` [here](https://github.com/fedimint/fedimint/blob/1d3758de7d6f8a0e6a466ee1f8e99cd65f0d90a5/fedimint-api/src/tiered_multi.rs#L132)

Reply:
I like the short form. But kept long snake form for consistency with the getter apis, because I don't see how to shorten two `get_x()` methods that take no params

Reply:
right, now I see what you mean. there is `get_balance` and `get_info`. hm, I guess we could do something like `pay(invoice)`, `fed_info()`, and `balance()`